File: River/River.Client/Assets/Packages/Ahoy.Net/Scripts/Sockets/SocketDataEventArgs.cs

using UnityEngine;
using System;
using System.Net;
using System.Net.Sockets;

namespace Ahoy.Net
{

    public struct SocketDataEventArgs
    {
        public AhoySocket receiver;
        public EndPoint sender;
        // public ArraySegment<byte> data;
        public ArraySegment<byte> data;

        public override string ToString()
        {
            return $"sender:{sender}, receiver: {receiver}, data length:{data.Count}, data:{data.ElementsToString()}";
        }
    }
}

File: River/River.Body/Ahoy/Sockets/SocketDataEventArgs.py


class SocketDataEventArgs():
    def __init__(self, sender, receiver, data):
        self.sender = sender
        self.receiver = receiver
        self.data = data

    def __str__(self):
        return "sender: {}, receiver: {}, data length: {} data: {}".format(self.sender, self.receiver, len(self.data), self.data)
# $"receiver: {receiver}, sender:{sender}, data length:{data.Count}, data:{data.ElementsToString()}";
